src/android/interoperability/cpp/cpp-bridge.md

```diff
@@ -0,0 +1,52 @@
+# C++ Bridge Declarations
+
+```rust,ignore
```

Comment:
As a more general tool, let us mark the untranslatable code snippets with

````suggestion

<!-- mdbook-xgettext: skip -->
```rust,ignore
````

This will completely avoid putting the code into the PO files and thus greatly reduce the workload of the translators.

Reply:
Please add such a comment before every code block which cannot be translated, including the ones that include their content from another file.

You can see the effect of this with

```
cargo install mdbook-i18n-helpers
MDBOOK_OUTPUT='{"xgettext": {"pot-file": "before.pot"}}' mdbook build -d po
MDBOOK_OUTPUT='{"xgettext": {"pot-file": "after.pot"}}' mdbook build -d po
diff po/before.pot po/after.pot
```

You can then regenerate the POT file periodically to check that it only contains text that can be meaningfully translated.

Reply:
Okay, can do! Which code blocks can't be translated, though? Do the translators usually translate any comments in code snippets?

Reply:
Things are moving fast :smile: We've just gotten a PR which will help a lot with the translation effort by only extracting comments and strings: https://github.com/google/mdbook-i18n-helpers/pull/109. This makes the large code blocks much less annoying for the translators.

To answer your questions:

> Okay, can do! Which code blocks can't be translated, though?

Code where all static strings are things like `"a: {a}"` from a format string.

> Do the translators usually translate any comments in code snippets?

It's a bit uneven today. The Simplified Chinese translation actually translated some comments: https://google.github.io/comprehensive-rust/zh-CN/hello-world/small-example.html (but not all, as far as I know).

Reply:
> This makes the large code blocks much less annoying for the translators.

Meaning: don't worry too much about this now.